models/classes/runner/RunnerToolStates.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA ;
+ */
+
+namespace oat\taoQtiTest\models\runner;
+
+trait RunnerToolStates
+{
+    abstract public function hasRequestParameter($name);
+
+    abstract public function getRequestParameter($name);
+
+    abstract public function getRawRequestParameter();
+
+    abstract public function getServiceLocator();
+
+    abstract public function getRunnerService();
+
+    abstract public function getServiceContext();
+
+    /**
+     * Save the tool state if some are found in the current request
+     *
+     * @throws \oat\oatbox\service\exception\InvalidServiceManagerException
+     * @throws \common_Exception
+     *
+     * @return boolean true if some state was found and saved
+     */
+    protected function saveToolStates()
+    {
+        $toolStateParameter = 'toolStates';
+        if ($this->hasRequestParameter($toolStateParameter)) {
```

Comment:
The `getRawRequestParameter` method calls `hasRequestParameter`, so there is no need to call it here. Instead we can assign the param value in a oneliner, and use one if statement, eg.:

```
$param = $this->getRawRequestParameter($toolStateParameter) ?: ''
if ($param) {
    // do stuff
}
```